debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -78,10 +81,9 @@ protected SnapshotContext prepare(ChangeEventSourceContext context) throws Excep
 
     @Override
     protected void connectionCreated(RelationalSnapshotContext snapshotContext) throws Exception {
-        LOGGER.info("Setting isolation level");
-        String transactionStatement = snapshotter.snapshotTransactionIsolationLevelStatement(slotCreatedInfo);
-        LOGGER.info("Opening transaction with statement {}", transactionStatement);
-        jdbcConnection.executeWithoutCommitting(transactionStatement);
+        if (snapshotter.shouldStreamEventsStartingFromSnapshot() && startingSlotInfo == null) {
```

Comment:
Where's the isolation level set then in case we _do_ the catch-up reading? Or asked in another way: why is it done differently for catch-up reading vs. not catch-up reading?

Reply:
Setting the isolation level is skipped here because it has already been set. For catch up streaming:
1. The snapshot transaction is opened and the current tx id is recorded. This transaction remains open.
2. Catch up streaming is performed until the tx id in the previous step is reached.
3. The snapshot starts. Since the snapshot transaction is already open and has executed statements, setting the isolation level is skipped.

Reply:
Ok, thanks. That makes sense. Could you put a comment along those lines into the code?

Reply:
Addressed this comment on #1627